### DISCUSSION TITLE
PCAP Recording - Showcase

### DISCUSSION THREAD

**Discussion by adamgeorge309** [inet-framework/inet#603] [Showcases]
Discuss the "PCAP Recording" showcase here. Ideas and recommendations are welcome.

**Comment by selzermauricio** [comment]
Is there any way to capture PCAP with the AODV protocol without the serialize UDP error ?
( Cannot serialize UDP header without turned off or properly computed CRC, try changing the value of crcMode parameter for Udp..)

**Comment by SandeepaDevin** [comment]
Cannot find serializer for VeinsInetSampleMessage -- in module (inet::Ipv4) Scenario.wifi[0].ipv4.ip
What cause this issue in in Pcaprecorder and how am I supposed to serialize this ?

**Comment by levy** [comment]
@selzermauricio The answer for your question is in the error message, add the following to your INI file: **.crcMode = "computed" and perhaps also **.fcsMode = "computed"

@SandeepaDevin You have to have a serializer for VeinsInetSampleMessage, see how the other serializers are implemented.

**Comment by Takarossa** [comment]
I'm sorry if this is a stupid question but is there a way to make your PCAP-recorder backwards compatible for INET 3.6.6?
I have changed the omnetpp.ini file so that the simulation can run and added the multivisualizer files. 

After those modifications the showcase simulation runs as expected and once it has concluded it generates the PCAP files containing IP, ICMP, TCP and UDP packets, but it seems like I can't capture any Ethernet frames, which is unfortunate since that's exactly what I need.

Judging from the comments in your code, I guess the issue could be related to these two lines:
**,crcMode = "computed"
**.fcsMode = "computed"

I have tried adding the files from inet 4.1.2 to inet 3.6.6 where you define this, but I haven't been able to get it to work.
It could also be the issue is that the lines containing:
*.hostX.pcapRecorder[*].pcapNetwork = Y 
are unrecognized. (X and Y are numbers)

I was wondering if you have any thoughts on whatever it's feasible to make this showcase work in INET 3.6.6?
I have attached my omnetpp.ini file in case anyone is curious. (OBS: Is as a txt file since github does not like the ini extension)
[omnetpp.txt](https://github.com/inet-framework/inet/files/9037987/omnetpp.txt)

**Comment by tuantitu22** [comment]
Hi, After run simulation of this example, I found pcap file of host2 has no data. I have tried to modified the code, but it didn't work.
The original code is:
*.host2.numPcapRecorders = 1
*.host2.pcapRecorder[*].packetFilter = "*ArpPacket"
